servicesExternal/web3_single.py

```diff
@@ -2,21 +2,31 @@
 from web3 import Web3 as WEB3
 
 from config import config
-
-
-class Web3Single:
+class Singleton(type):
+    _instances = {}
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        #not sure about should we use else or not
+        else:
+            cls._instances[cls].__init__(*args, **kwargs)
+        return cls.__instance
+
+
+class Web3Single(metaclass = Singleton):
     def __init__(self, web3Provider: Any = None, networkId: int = None):
-        pass
+        self.web3 = new WEB3(web3Provider or new WEB3.providers.HttpProvider(config.ethereum.nodeUrlDefault[config.ethereum.default])) #put here right link
+        self.networkId = networkId ? Web3Single.getNetworkName(networkId) : config.ethereum.default
```

Comment:
The `?` doesn't work in Python. See here: https://stackoverflow.com/questions/394809/does-python-have-a-ternary-conditional-operator